### PR TITLE
Fix hovertext bug

### PR DESCRIPTION
**Pull request by lewtun**
Figure update also requires an update of the color by column buttons

#### Reference Issues/PRs

Fixes #120

#### What does this implement/fix? Explain your changes.
Fixes bug where the hover text would not display for all nodes upon a change of the interactive mapper parameters. The source of the problem was that I overlooked the need to update a key plotly variable during the update. Note that the solution also requires us to update the color buttons since they depend on the node colours.

### CODE REVIEW

File: giotto/mapper/visualization.py [hunk 1]
```diff
@@ -277,12 +281,37 @@ def on_parameter_change(change):
 
             logger.info("Updating figure ...")
             with fig.batch_update():
-                node_trace, edge_trace, _, _, _ = \
+                node_trace, edge_trace, node_elements, node_colors, plot_options = \
                     _calculate_graph_data(
                         pipe, data, layout, layout_dim,
                         color_variable, node_color_statistic,  plotly_kwargs
```
Comment (ulupo) [inline giotto/mapper/visualization.py]: Two spaces before "plotly_kwargs"

**Comment by ulupo** [comment]
Just to understand the point on updating the color buttons: are we saying that this was not being done until now? And hence was another bug we had not noticed?

File: giotto/mapper/visualization.py (hunk 1 above)
Comment (ulupo) [reply giotto/mapper/visualization.py]: Fixed.

Comment (ulupo) [reply giotto/mapper/visualization.py]: Also fixed length-related linting problem due to the edited line above.

Comment (lewtun) [reply giotto/mapper/visualization.py]: Thanks!

**Comment by lewtun** [comment]
Yes, unfortunately this dropdown bug escaped my attention because in the interactive mode I had focused on seeing how the graph changed with parameter changes and did not make that extra step to also toggle the column coloring (see figure, where black markers denote wrong coloring).

Screen Shot 2020-01-10 at 09 37 27

**Comment by ulupo** [comment]
Thanks for the explanation! This is now ready to merge.

**Pull request merged**